coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -487,6 +489,84 @@ protected SqlNode visitGroupBy(ASTNode node, ParseContext ctx) {
     return ctx.grpBy;
   }
 
+
+
+  @Override
+  protected  SqlNode visitGroupingSets(ASTNode node,  ParseContext ctx){
+    // When Hive recognizes a grouping set, it omits the group by node in the constructed AST.
+    // However, Calcite requires the grouping set to be within a group by node when building SqlNode.
+    // Therefore, we need to add a group by node.
+    if(ctx.grpBy == null){
+      ctx.grpBy = new SqlNodeList(new ArrayList<SqlNode>(), ZERO);
+    }
+
+    List<SqlNode> groupingSets = visitChildren(node, ctx);
+    List<SqlNode> operands = new ArrayList<>();
+    if (groupingSets.isEmpty()) {
+      operands.add(CalciteUtil.createSqlNodeList(Collections.emptyList()));
+    } else  {
+      // In Hive, fields used in GROUPING SETS must be declared after the GROUP BY clause.
+      // However, when constructing Calcite SqlNode, the SqlIdentifier after GROUP BY isn't required;
+      // only the child nodes of TOK_GROUPING_SETS are needed, so they are filtered out.
+      List<SqlNode> operand = groupingSets.stream()
+              .filter(f -> !(f instanceof SqlIdentifier))
+              .collect(Collectors.toList());
+      operands.add(new SqlNodeList(operand, ZERO));
+    }
+    SqlNode groupingSetsNode = GROUPING_SETS.createCall(ZERO, operands);
+    ctx.grpBy.add(groupingSetsNode);
+
+    return groupingSetsNode;
+  }
+
+  @Override
+  protected  SqlNode visitGroupingSetsExpression(ASTNode node, ParseContext ctx){
+    List<SqlNode> identifiers = visitChildren(node, ctx);
+    if (identifiers == null || identifiers.isEmpty()){
+      return  CalciteUtil.createSqlNodeList(Collections.emptyList());
+    }else{
+      return ROW.createCall(ZERO, new SqlNodeList(identifiers, ZERO));
+    }
+  }
+
+  @Override
+  protected  SqlNode visitCubeGroupBy(ASTNode node, ParseContext ctx){
+    if(ctx.grpBy == null){
+      ctx.grpBy = new SqlNodeList(new ArrayList<SqlNode>(), ZERO);
+    }
```

Comment:
Could you add a similar comment on the relationship between the two representations? Here and in `RollUp`.